articles/imbalanced-data-in-ml/index.md

```diff
@@ -0,0 +1,126 @@
+
+#### Ensemble Learning
+
+An ensemble-based method can be used to deal with imbalanced datasets. The belief is that multiple learning methods are more effective than a single one. It is an approach that combines the performance or results of many classifiers to better the performance of a single classifier.
+
+Let’s briefly define a couple of ensemble methods;
+
+**Bagging**– This method attempts to apply similar learners on tiny sample populations then takes the mean of all predictions made.
+
+![bagging](/engineering-education/imbalanced-data-in-ml/bagging.png)
+
+Bagging
+
+[Source](https://www.analyticsvidhya.com/blog/2015/08/introduction-ensemble-learning/)
+
+**Boosting**– a technique that adjusts an observation's weight based on the most recent classification. Boosting attempts to increase the weight of an observation that has been incorrectly classified. This is an iterative process since new models are then trained on such inefficiencies of prior models. This makes the new models better and stronger than the previous ones. The resulting ensemble has many machine learning models. These models boast different accuracies and can provide better accuracies when used together. Boosting reduces bias error of models as well.
+
+#### The Right Evaluation Metrics
+
+Whenever there is an imbalance in class labels, the classification accuracy metric is not ideal for model performance. Here’s what we mean. Classification accuracy is a good metric when the samples belonging to each class are equal in number. Consider a scenario with 93% samples of class C and 7% class F in a training set. A model can very simply achieve 93% training accuracy by predicting each training sample in class C. This is even assuming that it fails to predict any samples in class F correctly.
+
+Thus, when dealing with imbalanced datasets, it is wise to use the correct evaluation metrics. It is not advisable to use accuracy as a measure of performance. We may consider metrics such as F1-score, precision, and recall.
+
+Here is an [article](https://towardsdatascience.com/metrics-to-evaluate-your-machine-learning-algorithm-f10ba6e38234) that defines the below metrics in greater detail.
+
+As defined in my previous [article](https://www.section.io/engineering-education/evaluating-ml-model-performance/):
+
+Precision is the number of true positives against the total positive results predicted by a classifier.
+
+Precision = True Positives / (True Positives + False Positives)
```

Comment:
Consider wrapping it in $$ Precision = \frac{True Positives}{True Positives + False Positives} $$ It makes it more professional. Consider doing the same for all the formulas mentioned.

Reply:
Math JAX file is needed at the end. @hectorkambow Your help is needed with this once the edits are made.